src/Controller/WelcomeController.php

```diff
@@ -37,6 +39,8 @@ public function index(Request $request): Response
         if ($form->isSubmitted() && $form->isValid()) {
             $game = $this->gameManipulator->createGame();
 
+            $this->session->set('player_hash', $game->getPlayer1()->getHash());
```

Comment:
You should manipule player session from your "player management" class.
The "player_hash" need to be in one place only